server/server.go

```diff
@@ -54,27 +58,58 @@ func (r *rodeServer) BatchCreateOccurrences(ctx context.Context, occurrenceReque
 	}, nil
 }
 
-func (r *rodeServer) AttestPolicy(ctx context.Context, request *pb.AttestPolicyRequest) (*pb.AttestPolicyResponse, error) {
-	log := r.logger.Named("AttestPolicy")
-	log.Debug("received requests")
+func (r *rodeServer) EvaluatePolicy(ctx context.Context, request *pb.EvaluatePolicyRequest) (*pb.EvaluatePolicyResponse, error) {
+	var err error
+	log := r.logger.Named("AttestPolicy").With(zap.String("policy", request.Policy), zap.String("resource", request.ResourceURI))
+	log.Debug("evaluate policy request received")
 
 	// check OPA policy has been loaded
+	err = r.opa.InitializePolicy(request.Policy)
+	if err != nil {
+		log.Error("error checking if policy exists", zap.Error(err))
+		return nil, status.Error(codes.Internal, "check if policy exists failed")
+	}
 
 	// fetch occurrences from grafeas
-	listOccurrencesResponse, err := r.grafeasCommon.ListOccurrences(ctx, &grafeas_proto.ListOccurrencesRequest{Filter: fmt.Sprintf("resource.uri = '%s'", request.ResourceURI)})
+	listOccurrencesResponse, err := r.grafeasCommon.ListOccurrences(ctx, &grafeas_proto.ListOccurrencesRequest{Parent: "projects/rode", Filter: fmt.Sprintf(`"resource.uri" == "%s"`, request.ResourceURI)})
 	if err != nil {
 		log.Error("list occurrences failed", zap.Error(err), zap.String("resource", request.ResourceURI))
 		return nil, status.Error(codes.Internal, "list occurrences failed")
 	}
+	log.Debug("Occurrences found", zap.Any("occurrences", listOccurrencesResponse))
 
 	// json encode occurrences. list occurrences response should not generate error
-	_, _ = protojson.Marshal(proto.MessageV2(listOccurrencesResponse))
+	input, _ := protojson.Marshal(proto.MessageV2(listOccurrencesResponse))
 
 	// evalute OPA policy
+	evaluatePolicyResponse, err := r.opa.EvaluatePolicy(request.Policy, input)
+	if err != nil {
+		log.Error("evaluate OPA policy failed")
```

Comment:
to clarify my earlier comment, this is where I would expect errors to be logged.  since this log statement exists, and `opa.EvaluatePolicy` also has a log statement, we're effectively logging it twice.

Reply:
I agree with this should be where the logging is done. The reason I also included the other logging statement is to get a stack trace to where the error actually happened instead of just there where it is handled. I think the problem is that go errors and this logger doesn't handle stack traces very well. It's a trade off between the code being cleaner and being able to easily troubleshoot errors. I'm on the fence.

Reply:
if we log the actual error, it should include a stack trace.  I'm wondering if we're doing something incorrectly with the `ClientError` implementation.

Reply:
Lets revisit this and come up with a logging / error handling strategy